hoomd/data/typeparam.py

```diff
@@ -191,10 +191,9 @@ def default(self):
     def default(self, value):
         self.param_dict.default = value
 
-    def _attach(self, cpp_obj, sim):
```

Comment:
@b-butler Is this change to the core HOOMD typeparameter API appropriate?

Reply:
@joaander to keep the current flexibility is it somewhat necessary since we do not know from the simulation the mesh types. I mentioned using the `ParameterDict` (`_param_dict`) for mesh potentials but @SchoeniPhlippsn wanted to keep the flexibility for multiple mesh types in the future.